packages/@aws-cdk/aws-autoscaling/lib/volume.ts

```diff
@@ -146,6 +148,13 @@ export class BlockDeviceVolume {
     return new this(undefined, `ephemeral${volumeIndex}`);
   }
 
+  /**
+   * Supresses a volume mapping
+   */
+  public static noDevice() {
+    return new this(undefined, 'NoDevice');
```

Comment:
Since `virtualName` is a part of the public API we can't really use it to signal `noDevice` (a user can set `virtualName = 'NoDevice'` which will result in an unexpected behavior), and since the `synthesizeBlockDeviceMappings` is implemented outside of the class we can't use a private member. To overcome the hurdles of the current implementation, we can use the `@interanl` annotation to define a public static member in `BlockDeviceVolume` that we will use a marker. In `BlockDeviceVolume`:

```javascript
/**
 * @internal
*/
public static _NO_DEVICE = new BlockDeviceVolume();

public static noDevice() {
   return this._NO_DEVICE;
}

```
And then in `synthesizeBlockDeviceMappings`:

```javascript
if (volume === BlockDeviceVolume._NO_DEVICE || mappingEnabled === false) {
  return {
    deviceName,
    noDevice: true,
  };
}
```
The `@interanl` annotation will cause the ts compiler to not emit the property in the declaration file.